langchain/document_loaders/evernote.py

```diff
@@ -5,78 +5,110 @@
         self.file_path = file_path
 
     def load(self) -> List[Document]:
-        """Load document from EverNote file."""
-        text = _parse_note_xml(self.file_path)
-        metadata = {"source": self.file_path}
-        return [Document(page_content=text, metadata=metadata)]
+        """Load documents from EverNote export file."""
+        return [
+            Document(
+                page_content=note["content"],
+                metadata={
+                    **{key: value for key, value in note.items() if key != "content"},
+                    **{"source": self.file_path}
+                })
+            for note in self._parse_note_xml(self.file_path) if note.get("content") is not None
+        ]
+
+    @staticmethod
+    def _parse_content(content: str) -> str:
+        try:
+            import html2text
+            return html2text.html2text(content)
+        except ImportError as e:
+            logging.error("Could not import `html2text`. Although it is not a required package to use Langchain, using the EverNote loader requires `html2text`. Please install `html2text` via `pip install html2text` and try again.")
+            raise e
+
+    @staticmethod
+    def _parse_resource(resource: list) -> dict:
+        rsc_dict: Dict[str, Any] = {}
+        for elem in resource:
+            if elem.tag == "data":
+                # Some times elem.text is None
+                rsc_dict[elem.tag] = b64decode(elem.text) if elem.text else b""
+                rsc_dict["hash"] = hashlib.md5(rsc_dict[elem.tag]).hexdigest()
+            else:
+                rsc_dict[elem.tag] = elem.text
+
+        return rsc_dict
+
+    @classmethod
+    def _parse_note(cls, note: List, prefix: Optional[str] = None) -> dict:
+        note_dict: Dict[str, Any] = {}
+        resources = []
+
+        def add_prefix(element_tag: str) -> str:
+            if prefix is None:
+                return element_tag
+            return f"{prefix}.{element_tag}"
+
+        for elem in note:
+            if elem.tag == "content":
+                note_dict[elem.tag] = cls._parse_content(elem.text)
+                # A copy of original content
+                note_dict["content-raw"] = elem.text
+            elif elem.tag == "resource":
+                resources.append(cls._parse_resource(elem))
+            elif elem.tag == "created" or elem.tag == "updated":
+                note_dict[elem.tag] = strptime(elem.text, "%Y%m%dT%H%M%SZ")
+            elif elem.tag == "note-attributes":
+                additional_attributes = cls._parse_note(elem, elem.tag)  # Recursively enter the note-attributes tag
+                note_dict.update(additional_attributes)
+            else:
+                note_dict[elem.tag] = elem.text
+
+        if len(resources) > 0:
+            note_dict["resource"] = resources
+
+        return {
+            add_prefix(key): value for key, value in note_dict.items()
+        }
+
+    @classmethod
```

Comment:
What do you think about using standalone functions instead? 

A developer will not have to worry about anything unexpected happening during inheritance (where identity of `cls` changes)

Reply:
Yes, good point. These should be static methods instead of class methods. As they only exist to support the EverNoteLoader class I think they should exist inside the class rather than outside in the file. Are you happy with them as static methods?

Reply:
I see your message above about using static methods instead so that's perfect. Have updated to be static methods.